extensions/data-plane/data-plane-aws-s3/src/test/java/org/eclipse/edc/connector/dataplane/aws/s3/arguments/S3DataPlaneIntegrationTestArgument.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2023 T-Systems International GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       T-Systems International GmbH
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.aws.s3.arguments;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class S3DataPlaneIntegrationTestArgument {
```

Comment:
Good point. I have overengineered it. 
The reason behind it is that the Named.of() method gets a T payload, which works similarly to passing the arguments object into the test method. This makes the test harder to read because I need to manually cast every object into its proper type.
```
    @ParameterizedTest
    @ArgumentsSource(S3DataPlaneIntegrationTestArgumentProvider.class)
    void shouldCopyFromSourceToSink(S3DataPlaneIntegrationTestArgument arguments) {
        var expectedKeyName = arguments.getExpectedKey();
        var expectedKeys = arguments.getExpectedKeys();
        var keys = arguments.getKeys();
        var body = arguments.getBody();
        var keyPrefix = arguments.getKeyPrefix();
     }
```
Instead of this:
```
    @ParameterizedTest
    @ArgumentsSource(S3DataPlaneIntegrationTestArgumentProvider.class)
    void shouldCopyFromSourceToSink(Arguments arguments) {
        var objects = arguments.get();
        String expectedKeyName = (String) objects[0];
        List<String> expectedKeys = (List<String>) objects[1];
        List<String> keys = (List<String>) objects[2];
        String body = (String) objects[3];
        String keyPrefix = (String) objects[4];
    }
```

Reply:
@Pluci you could have done just like:
```java
    @ParameterizedTest
    @ArgumentsSource(S3DataPlaneIntegrationTestArgumentProvider.class)
    void shouldCopyFromSourceToSink(String expectedKey, List<String> expectedKeys, List<String> keys, String body, String keyPrefix) {
     }
```

Because junit already binds arguments to the method parameters following the order.